Codecompanion should enable Treesitter Yaml

### ⚠️ Please verify that this bug has NOT been reported before.

- [x] I checked all existing issues and didn't find a similar issue

### Description

While trying out the Codecompaion module with a fresh and somewhat empty configuration of nvf I came across an error regarding "codecompanion chat" which refused to work without explicitly defining `languages.yaml.enable` and the global treesitter support.

### 👟 Reproduction steps

- Fresh empty configuration of nvf
- Define global LSP + Treesitter
- Define Codecompanion Module
- Build & Switch
- Try opening `:CodeCompanionChat`

### 👀 Expected behavior

The CodeCompanion Chat window opens

### 😓 Actual Behavior

I am getting an error message: `Could not find the Yaml Tree-sitter parser` 

### 💻 Metadata

 - system: `"x86_64-linux"`  - host os: `Linux 6.14.6, NixOS, 25.05 (Warbler), 25.05.20250510.d89fc19`  - multi-user?: `yes`  - sandbox: `yes`  - version: `nix-env (Nix) 2.28.3`  - channels(root): `"nixos"`  - nixpkgs: `/nix/store/ccfwxygjrarahgfv5865x2f828sjr5h0-source`

### 📝 Relevant log output

```bash
[ERROR] 2025-05-18 01:36:02
Could not find the Yaml Tree-sitter parser
Press ENTER or type command to continue
```

--- a/modules/plugins/assistant/codecompanion/config.nix
+++ b/modules/plugins/assistant/codecompanion/config.nix
@@ -21,7 +21,17 @@ in {
         };
       };
 
-      treesitter.enable = true;
+      treesitter = {
+        enable = true;
+
+        # Codecompanion depends on the YAML grammar being added. Below is
+        # an easy way of adding an user-configurable grammar package exposed
+        # by the YAML language module *without* enabling the whole YAML language
+        # module. The package is defined even when the module is disabled.
+        grammars = [
+          config.vim.languages.yaml.treesitter.package
+        ];
+      };
 
       autocomplete.nvim-cmp = {
         sources = {codecompanion-nvim = "[codecompanion]";};

